File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt for height of the pyramid
    int h;
    do
    {
        h = get_int("Height: ");
    }
    while (h < 1 || h > 8);

    // Building the pyramid
    for (int i = 1; i <= h; i++)
    {
        // Whitepaces before each row
        for (int j = h; j > i; j--)
        {
            printf(" ");
        }
        // Each row of the pyramid
        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        printf("  ");

        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: pset6/mario/mario.py
from cs50 import get_float, get_int, get_string

# Prompt user for height of the pyramid
h = 0

while True:
    if h > 0 and h < 9:
        # how that damn range is working?!
        for i in range(1, h + 1):
            for j in range(h - i, 0, -1):
                print(" ", end="")
            for j in range(i):
                print("#", end="")
            print("  ", end="")
            for j in range(i):
                print("#", end="")
            print("")
        break
    else:
        h = get_int("height: ")


# while True:
#     if h > 0 and h < 9:
#         # how that damn range is working?!
#         for i in range(0, h):
#             for j in range(i - h + 1, 0):
#                 print(" ", end="")
#             print((i + 1) * "#")
#         break
#     else:
#         h = get_int("height: ")

